tests/unit/timestamp_test.py

```
@@ -0,0 +1,220 @@
+def test_from_date_max_microseconds():
+    """Test from_date with maximum microseconds to ensure nanos calculation is correct."""
+    dt = datetime(2020, 1, 1, 0, 0, 0, 999999, tzinfo=timezone.utc)
+    ts = Timestamp.from_date(dt)
+
+    expected = 999_999_000
+    assert abs(ts.nanos - expected) < 1_000
+
+@pytest.mark.parametrize("bad_input", [None, [], {}, 3.14])
+def test_from_date_invalid_type(bad_input):
+    """Ensure from_date raises ValueError for invalid input types."""
+    with pytest.raises(ValueError):
```

Ruff (0.14.10)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)